xml_converter/tests/attribute_testing.py

```diff
@@ -0,0 +1,140 @@
+from dataclasses import dataclass
+import subprocess
+import re
+import os
+from typing import Dict, List
+
+# Path to compiled C++ executable
+xml_converter: str = "../build/xml_converter"
+
+
+@dataclass
+class Attribute_Tests:
+    attribute_name: str
+    attribute_path: str
+    tests: List[str]
+    output_result: Dict[str, List[str]]
+    expect_output_to_be_equal: Dict[str, bool]
+    expect_files_to_be_equal: Dict[str, bool]
+
+
+def run_xml_converter(*args):
+    try:
+        # Build the command to execute the C++ program with the desired function and arguments
+        cmd = [xml_converter] + list(args)
+
+        # Run the C++ program and capture its output
+        result = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+
+        # Check if the C++ program executed successfully
+        if result.returncode == 0:
+            return result.stdout
+        else:
+            return result.stderr
+    except Exception as e:
```

Comment:
If you know exactly what exceptions you are catching this is fine, but catching all exceptions can bite you.

Reply:
Good point. Since I don't have a specific one in mind that could happen, I'm going to remove this for now. We actually expect some errors if we test for errors.